Home assistant 0.111.1 Nuki update problem

## The problem
After update to 0.111.1 my nuki lock do not update status.

## Environment
host_os | HassOS 4.10
os_version | 4.19.126-v7
python_version | 3.7.7
supervisor | 227
version | 0.111.1

- Home Assistant Core release with the issue: 
- Last working Home Assistant Core release (if known): 0.110.7
- Operating environment (Home Assistant/Supervised/Docker/venv): HASS.io
- Integration causing this issue: 
- Link to integration documentation on our website: 

## Problem-relevant `configuration.yaml`
<!--
  An example configuration that caused the problem for you. Fill this out even
  if it seems unimportant to you. Please be sure to remove personal information
  like passwords, private URLs and other credentials.
-->

```yaml

```

## Traceback/Error logs

logs:
Logger: homeassistant.helpers.entity
Source: components/nuki/lock.py:124
First occurred: 11:40:21 (377 occurrences)
Last logged: 14:56:51

Update for lock.porta fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 279, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 472, in async_device_update
    await self.hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
    self._nuki_device.update(aggressive=level)
  File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
    "Failed to update data for lock. "
AssertionError: Failed to update data for lock. Nuki ID None volatized.
## Additional information

found also https://github.com/pschmitt/pynuki/issues/54 but my problem is not with netgear

> AssertionError: Failed to update data for lock. Nuki ID None volatized

Yeah, I see that also in my logs. The question is what changed on lib... @pschmitt ?

Same error here, starting from 111.1

`Logger: homeassistant.helpers.entity
Source: components/nuki/lock.py:124
First occurred: 20:06:31 (266 occurrences)
Last logged: 22:23:07

Update for lock.porta_d_ingresso fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 279, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 472, in async_device_update
    await self.hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
    self._nuki_device.update(aggressive=level)
  File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
    "Failed to update data for lock. "
AssertionError: Failed to update data for lock. Nuki ID None volatized.`

Joining the group of people that encounters this issue.

```
Logger: homeassistant.helpers.entity
Source: components/nuki/lock.py:124
First occurred: 9:36:53 AM (38 occurrences)
Last logged: 9:56:29 AM

Update for lock.haustur fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 279, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 472, in async_device_update
    await self.hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
    self._nuki_device.update(aggressive=level)
  File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
    "Failed to update data for lock. "
AssertionError: Failed to update data for lock. Nuki ID None volatized.
```

After some update (Nuki fw?) i have many problems with Nuki. Sometimes Nuki component is not available (ok after restart again for few hours). For example right now component IS available but LOCK command fail with this error in log:

```
2020-06-22 21:54:03 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event system_log_event[L]: name=homeassistant.helpers.entity, message=['Update for lock.muj_nuki_zamek fails'], level=ERROR, source=['components/nuki/lock.py', 124], timestamp=1592855643.219114, exception=Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 279, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 472, in async_device_update
    await self.hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
    self._nuki_device.update(aggressive=level)
  File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
    "Failed to update data for lock. "
AssertionError: Failed to update data for lock. Nuki ID None volatized.
, count=1, first_occurred=1592855643.219114>
```

HA 112.1
ERROR
Logger: homeassistant.helpers.entity
Source: components/nuki/lock.py:124
First occurred: 15:13:13 (3 occurrences)
Last logged: 15:14:15

Update for lock.alexanna fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update
    self.update  # type: ignore
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
    self._nuki_device.update(aggressive=level)
  File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
    "Failed to update data for lock. "
AssertionError: Failed to update data for lock. Nuki ID None volatized.

Same problem here. Started wit HA 111.1 (I think):

`Logger: homeassistant.helpers.entity
Source: components/nuki/lock.py:124
First occurred: 3. Juli 2020, 20:20:15 (2353 occurrences)
Last logged: 16:57:40

Update for lock.haustur fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update
    self.update  # type: ignore
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
    self._nuki_device.update(aggressive=level)
  File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
    "Failed to update data for lock. "
AssertionError: Failed to update data for lock. Nuki ID None volatized.
`

I have the same problem with the following logged:

`Logger: homeassistant.helpers.entity
Source: components/nuki/lock.py:124
First occurred: 10 July 2020, 14:20:03 (12392 occurrences)
Last logged: 0:51:07

Update for lock.home fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update
    self.update  # type: ignore
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
    self._nuki_device.update(aggressive=level)
  File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
    "Failed to update data for lock. "
AssertionError: Failed to update data for lock. Nuki ID None volatized.`

**HA details:**
arch | x86_64
chassis | vm
dev | false
docker | true
docker_version | 19.03.8
hassio | true
host_os | HassOS 4.8
installation_type | Home Assistant OS
os_name | Linux
os_version | 5.4.43
python_version | 3.7.7
supervisor | 228
timezone | Europe/Zurich
version | 0.112.3
virtualenv | false

**Nuki details:**
firmware: 1.9.6 (newest available for me)

I have the same problem. I haven't such problems before. Home Assistant and nuki work**ed** perfect.

Nuki 2.7.30
HA 112.4
ERROR
Logger: homeassistant.helpers.entity
Source: components/nuki/lock.py:124

2020-07-14 17:16:04 ERROR (MainThread) [homeassistant.helpers.entity] Update for lock.haupteingang fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update
    self.update  # type: ignore
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
    self._nuki_device.update(aggressive=level)
  File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
    "Failed to update data for lock. "
AssertionError: Failed to update data for lock. Nuki ID None volatized.
2020-07-14 17:16:35 ERROR (MainThread) [homeassistant.helpers.entity] Update for lock.haupteingang fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update
    self.update  # type: ignore
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
    self._nuki_device.update(aggressive=level)
  File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
    "Failed to update data for lock. "
AssertionError: Failed to update data for lock. Nuki ID None volatized.
2020-07-14 17:17:05 ERROR (MainThread) [homeassistant.helpers.entity] Update for lock.haupteingang fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update
    self.update  # type: ignore
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
    self._nuki_device.update(aggressive=level)
  File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
    "Failed to update data for lock. "
AssertionError: Failed to update data for lock. Nuki ID None volatized.

It’s absurd that no one is working on this bug.

Inviato da iPad

> Il giorno 15 lug 2020, alle ore 08:00, myGithub-Markus <notifications@github.com> ha scritto:
> 
> ﻿
> I have the same problem. I haven't such problems before. Home Assistant and nuki worked perfect.
> 
> Nuki 2.7.30
> HA 112.4
> ERROR
> Logger: homeassistant.helpers.entity
> Source: components/nuki/lock.py:124
> 
> 2020-07-14 17:16:04 ERROR (MainThread) [homeassistant.helpers.entity] Update for lock.haupteingang fails
> Traceback (most recent call last):
> File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state
> await self.async_device_update()
> File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update
> self.update # type: ignore
> File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
> result = self.fn(*self.args, **self.kwargs)
> File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
> self._nuki_device.update(aggressive=level)
> File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
> "Failed to update data for lock. "
> AssertionError: Failed to update data for lock. Nuki ID None volatized.
> 2020-07-14 17:16:35 ERROR (MainThread) [homeassistant.helpers.entity] Update for lock.haupteingang fails
> Traceback (most recent call last):
> File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state
> await self.async_device_update()
> File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update
> self.update # type: ignore
> File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
> result = self.fn(*self.args, **self.kwargs)
> File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
> self._nuki_device.update(aggressive=level)
> File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
> "Failed to update data for lock. "
> AssertionError: Failed to update data for lock. Nuki ID None volatized.
> 2020-07-14 17:17:05 ERROR (MainThread) [homeassistant.helpers.entity] Update for lock.haupteingang fails
> Traceback (most recent call last):
> File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state
> await self.async_device_update()
> File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update
> self.update # type: ignore
> File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
> result = self.fn(*self.args, **self.kwargs)
> File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update
> self._nuki_device.update(aggressive=level)
> File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update
> "Failed to update data for lock. "
> AssertionError: Failed to update data for lock. Nuki ID None volatized.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

> 
> It’s absurd that no one is working on this bug. Inviato da iPad
> […](#)
> Il giorno 15 lug 2020, alle ore 08:00, myGithub-Markus ***@***.***> ha scritto: ﻿ I have the same problem. I haven't such problems before. Home Assistant and nuki worked perfect. Nuki 2.7.30 HA 112.4 ERROR Logger: homeassistant.helpers.entity Source: components/nuki/lock.py:124 2020-07-14 17:16:04 ERROR (MainThread) [homeassistant.helpers.entity] Update for lock.haupteingang fails Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state await self.async_device_update() File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update self.update # type: ignore File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run result = self.fn(*self.args, **self.kwargs) File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update self._nuki_device.update(aggressive=level) File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update "Failed to update data for lock. " AssertionError: Failed to update data for lock. Nuki ID None volatized. 2020-07-14 17:16:35 ERROR (MainThread) [homeassistant.helpers.entity] Update for lock.haupteingang fails Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state await self.async_device_update() File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update self.update # type: ignore File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run result = self.fn(*self.args, **self.kwargs) File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update self._nuki_device.update(aggressive=level) File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update "Failed to update data for lock. " AssertionError: Failed to update data for lock. Nuki ID None volatized. 2020-07-14 17:17:05 ERROR (MainThread) [homeassistant.helpers.entity] Update for lock.haupteingang fails Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 272, in async_update_ha_state await self.async_device_update() File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 466, in async_device_update self.update # type: ignore File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run result = self.fn(*self.args, **self.kwargs) File "/usr/src/homeassistant/homeassistant/components/nuki/lock.py", line 124, in update self._nuki_device.update(aggressive=level) File "/usr/local/lib/python3.7/site-packages/pynuki/device.py", line 90, in update "Failed to update data for lock. " AssertionError: Failed to update data for lock. Nuki ID None volatized. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

You mean, with the absurd amounts of monthly support fees we are paying for this open-source project? Agree!

I am not sure which module is causing the error (HA, pynuki or nuki itself). Fact is the initialization works. All devices (locks and openers) are found via the nuki bridge and updates a couple of time their properties. After a time of x hours the function updating properties of a lock (it would be interesting if this also happens with an opener? ... I have only one lock) fails in line 86 of pynuki `for l in self._bridge._get_devices(self.device_type) if l.nuki_id == self.nuki_id`. Whatever the reason is, this does not work and get back a empty result of **l** and throw's this error with `self.nuki_id` none which is shown in the ha log as `AssertionError: Failed to update data for lock. Nuki ID **None** volatized.`

> 
> 
> I am not sure which module is causing the error (HA, pynuki or nuki itself). Fact is the initialization works. All devices (locks and openers) are found via the nuki bridge and updates a couple of time their properties. After a time of x hours the function updating properties of a lock (it would be interesting if this also happens with an opener? ... I have only one lock) fails in line 86 of pynuki/device.py `for l in self._bridge._get_devices(self.device_type) if l.nuki_id == self.nuki_id`. Whatever the reason is, this does not work and get back a empty result of **l** and throw's this error with `self.nuki_id` **none** which is shown in the ha log as `AssertionError: Failed to update data for lock. Nuki ID **None** volatized.`

I have both the Lock and the Opener. My Opener has still worked perfectly fine for the full uptime of HA, only the Lock stops functioning. If this helps.

--- a/CODEOWNERS
+++ b/CODEOWNERS
@@ -280,7 +280,7 @@ homeassistant/components/notion/* @bachya
 homeassistant/components/nsw_fuel_station/* @nickw444
 homeassistant/components/nsw_rural_fire_service_feed/* @exxamalte
 homeassistant/components/nuheat/* @bdraco
-homeassistant/components/nuki/* @pvizeli
+homeassistant/components/nuki/* @pschmitt @pvizeli
 homeassistant/components/numato/* @clssn
 homeassistant/components/nut/* @bdraco
 homeassistant/components/nws/* @MatthewFlamm

--- a/homeassistant/components/nuki/manifest.json
+++ b/homeassistant/components/nuki/manifest.json
@@ -2,6 +2,10 @@
   "domain": "nuki",
   "name": "Nuki",
   "documentation": "https://www.home-assistant.io/integrations/nuki",
-  "requirements": ["pynuki==1.3.7"],
-  "codeowners": ["@pvizeli"]
+  "requirements": [
+    "pynuki==1.3.8"
+  ],
+  "codeowners": [
+    "@pvizeli"
+  ]
 }

--- a/homeassistant/components/nuki/manifest.json
+++ b/homeassistant/components/nuki/manifest.json
@@ -2,10 +2,6 @@
   "domain": "nuki",
   "name": "Nuki",
   "documentation": "https://www.home-assistant.io/integrations/nuki",
-  "requirements": [
-    "pynuki==1.3.8"
-  ],
-  "codeowners": [
-    "@pvizeli"
-  ]
+  "requirements": ["pynuki==1.3.8"],
+  "codeowners": ["@pschmitt", "@pvizeli"]
 }

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -1497,7 +1497,7 @@ pynetgear==0.6.1
 pynetio==0.1.9.1
 
 # homeassistant.components.nuki
-pynuki==1.3.7
+pynuki==1.3.8
 
 # homeassistant.components.nut
 pynut2==2.1.2

